packages/databyss-ui/editor/slate/page/reducer.js

```diff
@@ -255,6 +258,41 @@ const deleteBlockById = id => (editor, value, next) => {
   next(editor, value)
 }
 
+/*
+updates all sources provided in the ID list
+*/
+const onUpdateSource = (source, idList) => (editor, value, next) => {
```

Comment:
Can we compute the idList here so we don't have to pass it in?

Reply:
the `idList` can be computed in the slate reducer but we will still need to pass the `blocks` property since the slate reducer does not have access to our state values. This can be computed with the editor state when the `refId` is attached to the `data` attribute of each node.